identity/identity.go

```diff
@@ -84,10 +92,35 @@ func (srv *Service) enrollDevice() (*domain.Enrollment, error) {
 
 	// Store the enrollment credentials
 	err = srv.storeCredentials(resp.Enrollment)
+	if err != nil {
+		return nil, err
+	}
+
+	// Store device data in a separate file
+	if resp.Enrollment.DeviceData != "" {
```

Comment:
[Optional] I've tended to use `len(s) != 0` instead of s != "" (though both forms are fine)